MathLib/Vector3.h

```diff
@@ -52,7 +52,7 @@ class TemplateVector3 : public MathLib::TemplatePoint<T>
     /**
      * Construct Vector3 from TemplatePoint.
      */
-    TemplateVector3(TemplatePoint<T,3> const& p) :
+    explicit TemplateVector3(TemplatePoint<T,3> const& p) :
```

Comment:
If I remember correctly marking this ctor `explicit` required more changes in the code.

Reply:
No, I don't.

Reply:
Maybe this is the reason why most of the tests are failing. Will revert the change.

Reply:
Drop the change. It's easier to switch to Eigen::Vector3d.